packages/dialect-react-ui/components/Notifications/index.tsx

```diff
@@ -35,111 +37,153 @@ interface NotificationsProps {
 }
 
 function InnerNotifications(props: NotificationsProps): JSX.Element {
-  const {
-    info: { apiAvailability },
-  } = useDialectSdk();
-  const { dappAddress } = useDialectDapp();
   const { isCreatingThread } = useThreads();
+  const { dappAddress } = useDialectDapp();
   const { thread, isDeletingThread, isFetchingThread } = useThread({
     findParams: { otherMembers: dappAddress ? [dappAddress] : [] },
   });
+  const { adapter: wallet } = useDialectWallet();
 
-  const isDialectAvailable = Boolean(thread);
-
-  const cannotDecryptDialect =
-    !apiAvailability.canEncrypt && thread?.encryptionEnabled;
+  const {
+    addresses: { WALLET: walletAddress },
+    isFetching: isFetchingAddresses,
+    create: createAddress,
+    delete: deleteAddress,
+    isCreatingAddress,
+    isDeletingAddress,
+  } = useAddresses();
+
+  const isWeb3Enabled =
+    walletAddress?.enabled ||
+    isCreatingThread ||
+    isCreatingAddress ||
+    isDeletingThread ||
+    isDeletingAddress;
 
-  const { isSigning, isEncrypting } = useDialectWallet();
+  const { scrollbar } = useTheme();
+  const { navigate } = useRoute();
+  const prevThread = usePrevious(thread);
 
-  const {
-    addresses: { wallet: walletObj },
-  } = useDialectCloudApi();
+  const showThread = useCallback(() => {
+    if (!thread) {
+      return;
+    }
+    () =>
+      navigate(RouteName.Thread, {
+        params: {
+          threadId: thread.id,
+        },
+      });
+  }, [navigate, thread]);
 
-  const [isSettingsOpen, setSettingsOpen] = useState(false);
-  const isWeb3Enabled = walletObj
-    ? Boolean(walletObj?.enabled)
-    : isDialectAvailable;
+  const showSettings = useCallback(() => {
+    () => navigate(RouteName.Settings);
+  }, [navigate]);
 
-  const toggleSettings = useCallback(
-    () => setSettingsOpen(!isSettingsOpen),
-    [isSettingsOpen, setSettingsOpen]
-  );
 
-  const { scrollbar } = useTheme();
+  // Sync state for web3 channel in case of errors
+  useEffect(
+    function syncState() {
+      if (thread && prevThread?.id.equals(thread.id)) {
+        return;
+      }
 
-  const { navigate } = useRoute();
+      if (
+        !wallet?.publicKey ||
+        isFetchingAddresses ||
+        isFetchingThread ||
+        isCreatingThread ||
+        isDeletingThread
+      )
+        return;
```

Comment:
could you please add a block statement for this `return`. Otherwise it looks like it's floating :D

Reply:
think we should ban this style in eslint haha